src/discordtipbot/discordbot.cr

```diff
@@ -500,13 +500,14 @@ class DiscordBot
       break if new_msgs.last.timestamp < before
     end
 
-    msgs.reject!(&.author.bot)
-    msgs.reject! { |x| x.timestamp < before }
-    msgs.reject! { |x| x.author.id == msg.author.id }
-
-    authors = Array(UInt64).new
-    msgs.each { |x| authors << x.author.id }
-    authors.uniq!
-    return authors
+    authors = Set(UInt64).new
+
+    msgs.map do |x|
+      next if x.author.bot
+      next if x.timestamp < before
+      authors.add(x.author.id)
+    end
+
+    return authors.delete(msg.author.id)
```

Comment:
You can remove the `return`, it's implicit.

Secondly, can't you just `next if x.author.id == msg.author.id` in the above iterations, and implicitly return `authors`?

Reply:
I wasn't sure wether it would be faster do the check once and remove the author's id once or to check often.

Reply:
I could write a benchmark to see but its probably negligible if I had to take a guess

Reply:
I think/guess removing once is faster